src/binary/non_blocking/raw_binary_reader.rs

```diff
@@ -1153,11 +1153,10 @@ impl<'a> Iterator for AnnotationsIterator<'a> {
         if remaining == 0 {
             return None;
         }
-        // This iterator cannot be created unless the reader is currently parked on a value.
-        // If the reader is parked on a value, the complete annotations sequence is in the buffer.
-        // Therefore, reading symbol IDs from this byte slice cannot fail. This allows us to safely
-        // unwrap the result of this `read` call.
-        let var_uint = VarUInt::read(&mut self.data).unwrap();
+        let var_uint = match self.data.read_var_uint() {
+            Ok(var_uint) => var_uint,
+            Err(e) => return Some(Err(e)),
+        };
```

Comment:
What drives the change here? Is the context about positioning or buffered bytes different now?

Reply:
The method `VarUInt::read` was deleted. The rest of this type uses `BinaryBuffer`'s read operations; this change was changing `self.data` from being a `io::Cursor(&[u8])` that we read from using `VarUInt::read()` to being a `BinaryBuffer<&'a [u8]>` that has its own read operation.

The invariant described in the comment is still true; I think I could have kept it but didn't want to do the deep dive needed to guarantee that.